### PR TITLE
Create package @gravatar/quick-editor

### PR DESCRIPTION
**Pull request by jcheringer**
## Proposed Changes

This PR introduces a new package: `@gravatar/quick-editor`

Below is a usage example:

```HTML
<img
	class="avatar"
	src="https://1.gravatar.com/avatar/[AVATAR_HASH]?size=256"
/>
<button id="button">Edit Avatar</button>
```
```JavaScript
import GravatarQuickEditor, { GravatarQuickEditorCore } from '@gravatar/quick-editor';

document.addEventListener( 'DOMContentLoaded', () => {
	new GravatarQuickEditor( {
		email: 'email@provider.com',
		scope: [ 'avatars' ],
		editorTriggerSelector: '#button',
		avatarSelector: '.avatar',
	} );

	// QuickEditorCore usage
	const quickEditorCore = new GravatarQuickEditorCore( {
		email: 'email@provider.com',
		scope: [ 'avatars', 'about' ],
		local: 'es',
		onProfileUpdated: ( type: ProfileUpdatedType ) => {
			console.log( type );
		},
	} );

	document.querySelector( '#button' )?.addEventListener( 'click', () => {
		quickEditorCore.open();
	} );
} );
```

The constructor expects the following parameters:

* `triggerSelector`: A selector to the element that triggers the popup to be opened
* `email`: The user's email associated with the identity to be edited
* `options`: Optional parameter with the following options:
  * `scope`: allows defining the sections we want to show in the Quick Editor.
  * `avatarSelector`: A selector to match the avatar images on the page

## Testing Instructions

* Checkout this branch and run `npm run start`
* The playground page should open
* Change the files inside the folder `web/packages/quick-editor/playground/` with your Gravatar profile data
* Click the "Edit Avatar" the Quick Editor should open
* Change your Avatar image
* The images in the playground should change as well

### CODE REVIEW

File: web/packages/quick-editor/src/index.ts
Comment (wellyshen) [inline web/packages/quick-editor/src/index.ts]: We could provide a more specific type for a better DX. 

```suggestion
	scope?: ( 'avatars' | 'about' | '...' )[];
```

File: web/packages/quick-editor/src/index.ts
Comment (wellyshen) [inline web/packages/quick-editor/src/index.ts]: The `is_quick_editor` won't work in the login flow.

```suggestion
			`https://gravatar.com/profile?email=${ this._email }&scope=${ this._scope.join( ',' ) }&is_quick_editor=true`,
```

File: web/packages/quick-editor/package.json
Comment (wellyshen) [inline web/packages/quick-editor/package.json]: The `@gravatar` has been used. Therefore, we're using `@gravatar-com` instead, see: https://github.com/Automattic/gravatar/blob/trunk/web/packages/hovercards/package.json#L2

File: web/packages/quick-editor/package.json
Comment (wellyshen) [inline web/packages/quick-editor/package.json]: I suggest we start with `0.x.0` or `1.0.0-alpha`, `1.0.0-beta`, etc. This approach gives us more room for API changes.

File: web/packages/quick-editor/webpack/config.core.js
Comment (wellyshen) [inline web/packages/quick-editor/webpack/config.core.js]: [The library name is already used by the hovercards](https://github.com/Automattic/gravatar/blob/trunk/web/packages/hovercards/webpack/config.core.js#L61). If both are used, there might be a conflict. Perhaps we could use `GravatarQuickEditor` instead?

File: web/packages/quick-editor/src/index.ts
Comment (wellyshen) [inline web/packages/quick-editor/src/index.ts]: I’m concerned this might be outputted by the developer’s site. From a DX perspective, ideally, our library's error logs should only be displayed in the development build / stage (as shown below). Unfortunately, we haven't handled this case yet 😅.

```js
if ( 'production' !== process.env.NODE_ENV ) {
  console.error( '...' );
}
```

**Comment by wellyshen** [comment]
@jcheringer Thank you for contributing to this open-source project! I'm excited to see a library that enables people worldwide to use Gravatar profiles. This is a fantastic library with great potential for expanding the use of Gravatar profiles. To ensure we offer a great DX, I'd like to start the review by focusing on the API design.

### 💡 Providing Open / Close Methods To Control The Popup Window

I was thinking we could offer an open method, e.g., `quickEditor.open()`, instead of setting [a single trigger element](https://github.com/Automattic/gravatar/pull/14/files#diff-4e695a48fe439fc72ae2bdb02049aca4bf495fbcc8a264d5315e75dcd1f984e2R13) and [handling the click event internally](https://github.com/Automattic/gravatar/pull/14/files#diff-4e695a48fe439fc72ae2bdb02049aca4bf495fbcc8a264d5315e75dcd1f984e2R38-R40). This method would allow the user to open the quick editor at any place and time point they choose.

### 💡 Providing Event Callbacks For Users To Perform Their Desired Tasks

I noticed that we can update the user's avatar by [parsing and modifying the img element's src attribute internally](https://github.com/Automattic/gravatar/pull/14/files#diff-4e695a48fe439fc72ae2bdb02049aca4bf495fbcc8a264d5315e75dcd1f984e2R49-R61). However, there are additional edge cases we need to consider, such as validating the Gravatar URL and checking if we are overriding the existing URL parameter.

I suggest we provide relevant events, such as `onProfileUpdateSuccess` and `onProfileUpdateError`, to allow developers to perform their desired actions. This way, we can avoid directly modifying the `img` elements.

### 💡 Offer a `local` option

We could offer a local option to allow users to [access the Quick Editor](https://github.com/Automattic/gravatar/pull/14/files#diff-4e695a48fe439fc72ae2bdb02049aca4bf495fbcc8a264d5315e75dcd1f984e2R73) in their preferred language.

### 💭 Here's My Idea For The Library

Based on the above points, I'd like to share my picture for the library. Please let me know your thoughts bro, thanks!

**API:**
```ts
const quickEditor = new GravatarQuickEditor( {
	email: '...', // If it is available at the time of initialization
	scope: [ 'avatars' ],
	local: 'en', // "en" by default
	onProfileUpdateSuccess: ( type ) => {
		// Developers can handle the success event here. E.g:
		// - Update avatars
		// - Call the profile REST API to update the data
		// - Close the popup window
		// - And more...
	},
	onProfileUpdateError: ( type ) => {
		// Developers can handle the error event here.
	},
	onOpened: () => {
		// (Good to have) Developers can handle the event when the popup window is opened.
	},
	onClosed: () => {
		// (Good to have) Developers can handle the event when the popup window is closed.
	},
} );

// Open the popup window at anytime and anywhere
quickEditor.open( email /* If it's available at the time of opening */ );

// Close the popup window at anytime and anywhere
quickEditor.close();
```

**Types:**
```ts
export type Scope = ( 'avatars' | 'profiles' | '...' )[];

export type OnProfileUpdateSuccess = ( type: string ) => void;

export type OnProfileUpdateError = ( type: string ) => void;

export type OnOpened = () => void; // Good to have

export type OnClosed = () => void; // Good to have

export type Options = Partial< {
	email: string;
	scope: Scope;
	local: string;
	onProfileUpdateSuccess: OnProfileUpdateSuccess;
	onProfileUpdateError: OnProfileUpdateError;
	onOpened: OnOpened; // Good to have
	onClosed: OnClosed; // Good to have
} >;

export type Open = ( email?: string ) => void;

export type Close = () => void;
```

File: web/packages/quick-editor/src/index.ts
Comment (wellyshen) [inline web/packages/quick-editor/src/index.ts]: Does it work for `https://zh-tw.gravatar.com`?

File: web/packages/quick-editor/src/index.ts
Comment (jcheringer) [reply web/packages/quick-editor/src/index.ts]: ✅

File: web/packages/quick-editor/src/index.ts
Comment (jcheringer) [reply web/packages/quick-editor/src/index.ts]: ✅

File: web/packages/quick-editor/package.json
Comment (jcheringer) [reply web/packages/quick-editor/package.json]: ✅

File: web/packages/quick-editor/package.json
Comment (jcheringer) [reply web/packages/quick-editor/package.json]: ✅

File: web/packages/quick-editor/webpack/config.core.js
Comment (jcheringer) [reply web/packages/quick-editor/webpack/config.core.js]: ✅

File: web/packages/quick-editor/src/index.ts
Comment (jcheringer) [reply web/packages/quick-editor/src/index.ts]: I didn't know there was a local string like that 😆 
I've adjusted it.

File: web/packages/quick-editor/src/index.ts
Comment (jcheringer) [reply web/packages/quick-editor/src/index.ts]: I have removed some of the error messages that may be unnecessary.
But I don't know if we should show them only on the developer build.
There are situations where a developer could get a hint of what's going wrong in their site if a user mentions an error message in the console.

**Comment by jcheringer** [comment]
Hey, @wellyshen! Thanks for the review and detailed explanation.

I've adjusted the overall structure to follow your suggestions.
We now have two primary classes for the developers to use:

`GravatarQuickEditorCore` has the core functionality, and it's close to what you suggested, allowing the developers more granular control over the functionalities.

`GravatarQuickEditor` aims to simplify and streamline developers' implementations. This class will set up event listeners based on the selectors passed to the constructor and auto-refresh the avatars on the page if the appropriate selector is given.

Let me know what you think.

File: web/packages/quick-editor/package.json
Comment (wellyshen) [inline web/packages/quick-editor/package.json]: Sorry bro, let's start it with `0.0.0`. Coz we will need to test the npm release and check if all the builds (i.e. esm, umd, etc.) are working as expected via the very minimal version (e.g. `0.0.1`, etc.) then.

File: web/packages/quick-editor/webpack/config.core.js
Comment (wellyshen) [inline web/packages/quick-editor/webpack/config.core.js:60]: After thinking twice... Maybe we could give it a better name (or shorthand?). Because it will be a global variable to host the relevant modules. Using the hovercards lib as an example: `Gravatar.Hovercards`. So for the Quick Editor it will be `GravatarQuickEditor.GravatarQuickEditor` (`GravatarQuickEditor.default` for now)

File: web/packages/quick-editor/src/index.ts
Comment (wellyshen) [inline web/packages/quick-editor/src/index.ts]: If I remember correctly. We need to give it a name for the `UMD` build to work as below:

- Current: [The_UMD_global_variable](https://github.com/Automattic/gravatar/pull/14/files#diff-de5746fe258fbb418c3d465c4960dd234f39ff0ca3947b6eb5612274eb96c0cbR60).default
- Suggest: [The_UMD_global_variable](https://github.com/Automattic/gravatar/pull/14/files#diff-de5746fe258fbb418c3d465c4960dd234f39ff0ca3947b6eb5612274eb96c0cbR60).GravatarQuickEditor

```suggestion
export { default as QuickEditor } from './quick-editor';
```

File: web/packages/quick-editor/src/quick-editor-core.ts
Comment (wellyshen) [inline web/packages/quick-editor/src/quick-editor-core.ts]: We missed a type for this method, please refer to: https://github.com/Automattic/gravatar/blob/trunk/web/packages/hovercards/src/core.ts#L504

File: web/packages/quick-editor/src/quick-editor-core.ts
Comment (wellyshen) [inline web/packages/quick-editor/src/quick-editor-core.ts:70]: We could check the `this._scope`'s values (and prevent the incorrect values are passed to gravatar.com) for developers, especially for non-ts developers.

File: web/packages/quick-editor/src/quick-editor-core.ts
Comment (wellyshen) [inline web/packages/quick-editor/src/quick-editor-core.ts]: Based on [the type definition](https://github.com/Automattic/gravatar/pull/14/files#diff-33471df6c68e702f88ed75df6024470aef629542105f2314645837be7e1721e8R5). Seems the event type should be a must.

File: web/packages/quick-editor/src/quick-editor.ts
Comment (wellyshen) [inline web/packages/quick-editor/src/quick-editor.ts:35]: [Should we verify the img element and the `src`](https://github.com/Automattic/gravatar/blob/trunk/web/packages/hovercards/src/core.ts#L145)?

File: web/packages/quick-editor/src/quick-editor.ts
Comment (wellyshen) [inline web/packages/quick-editor/src/quick-editor.ts:36]: I guess we will need to mention the added `t` param on our doc then.

File: web/packages/quick-editor/src/quick-editor.ts
Comment (wellyshen) [inline web/packages/quick-editor/src/quick-editor.ts]: Q: Do you think the `1000` delay could be an option with the default as `1000`?

File: web/packages/quick-editor/src/quick-editor-core.ts
Comment (wellyshen) [inline web/packages/quick-editor/src/quick-editor-core.ts:16]: Great job 💙

File: web/packages/quick-editor/src/quick-editor.ts
Comment (wellyshen) [inline web/packages/quick-editor/src/quick-editor.ts]: Q: Should the  `editorTriggerSelector` be a must parameter to use the library?

```suggestion
	constructor( editorTriggerSelector, { email, avatarSelector, scope, local }: QuickEditorOptions ) {
```

**Review by wellyshen — COMMENTED** [review]
Hey bro, I think the library architecture is great now. Awesome job! 🤩

### For the core module

- Do you think it's worth providing a `close` method (and the related event) for developers? The use case I can think of is allowing developers to close the popup window when the user opens other modals / popups or leaves the page, etc.

### For the Quick Editor module

- Currently, the way we use it is to initialize the class after developers have gotten the user's email. Do you think we could provide an `init( email?: string )` (or `run( email?: string )`) method for developers to run the relevant code at the desired time?

```ts
// Run it immediately
new QuickEditor( 'some-selector', { email: 'test@gmail.com' } ).init();

// Run it lazily
const quickEditor = new QuickEditor( ... );
quickEditor.init( 'test@gmail.com' );
```

### Next

- Make the relevant development things work, e.g. `package.json` scripts, etc.

**Comment by jcheringer** [comment]
Thanks for the second round of review, @wellyshen!
Answering to some of your questions:

> Do you think it's worth providing a close method (and the related event) for developers? The use case I can think of is allowing developers to close the popup window when the user opens other modals / popups or leaves the page, etc.

I'm not sure we should allow a developer to close the popup. Imagine a user is using the Quick Editor, and out of nowhere, the popup closes because the developer called the close method. That would be a poor experience.

> Currently, the way we use it is to initialize the class after developers have gotten the user's email. Do you think we could provide an init( email?: string ) (or run( email?: string )) method for developers to run the relevant code at the desired time?

I don't think that's the purpose of that class. If a developer wants more control, they could use the `GravatarQuickEditorCore`.

> Make the relevant development things work, e.g. package.json scripts, etc.

Maybe that's something we could address in a follow-up PR?

File: web/packages/quick-editor/src/index.ts
Comment (jcheringer) [reply web/packages/quick-editor/src/index.ts]: My idea here was to make the `GravatarQuickEditor` available as the default module, allowing developers to import it like `import GravatarQuickEditor from '@gravatar-com/quick-editor';
But that's not mandatory, I just thought it would make sense to have a default module.

File: web/packages/quick-editor/src/quick-editor-core.ts
Comment (jcheringer) [reply web/packages/quick-editor/src/quick-editor-core.ts:70]: I'm not sure we need to worry about that. 
An incorrect value would be ignored by the profile editor as it wouldn't match any of the routes.

File: web/packages/quick-editor/src/quick-editor.ts
Comment (jcheringer) [reply web/packages/quick-editor/src/quick-editor.ts:35]: The `URL.canParse( avatarElement.src )` right above seems enough as a safety check.

File: web/packages/quick-editor/src/quick-editor.ts
Comment (jcheringer) [reply web/packages/quick-editor/src/quick-editor.ts:36]: Yes, we can mention that.

File: web/packages/quick-editor/webpack/config.core.js
Comment (jcheringer) [reply web/packages/quick-editor/webpack/config.core.js:60]: In this case, couldn't we keep `Gravatar` then?
Would that override the `Gravatar.Hovercards` if both packages are imported?

File: web/packages/quick-editor/src/quick-editor.ts
Comment (jcheringer) [reply web/packages/quick-editor/src/quick-editor.ts]: The TypeScript definition makes it mandatory. It wouldn't trigger an error in a non-TS environment, but IMO, it feels cleaner to have everything as an object property.
It also keeps both `GravatarQuickEditor` and `GravatarQuickEditorCore` initialization similar.

File: web/packages/quick-editor/package.json
Comment (jcheringer) [reply web/packages/quick-editor/package.json]: ✅

File: web/packages/quick-editor/src/quick-editor-core.ts
Comment (jcheringer) [reply web/packages/quick-editor/src/quick-editor-core.ts]: ✅

File: web/packages/quick-editor/src/quick-editor-core.ts
Comment (jcheringer) [reply web/packages/quick-editor/src/quick-editor-core.ts]: ✅

File: web/packages/quick-editor/src/quick-editor.ts
Comment (jcheringer) [reply web/packages/quick-editor/src/quick-editor.ts]: ✅

File: web/packages/quick-editor/webpack/config.core.js
Comment (wellyshen) [reply web/packages/quick-editor/webpack/config.core.js:60]: > Would that override the Gravatar.Hovercards if both packages are imported?

I think the previous `Gravatar` will be override by the later one.

File: web/packages/quick-editor/src/quick-editor-core.ts
Comment (wellyshen) [reply web/packages/quick-editor/src/quick-editor-core.ts:70]: Yes, but JS developers may silently fail to get the correct scope. We can do the enhancement in the next iteration.

File: web/packages/quick-editor/src/quick-editor.ts
Comment (wellyshen) [reply web/packages/quick-editor/src/quick-editor.ts:35]: Since we're helping users update their Gravatar's avatars. Do you think it's worth checking if the `src` belongs to a Gravatar URL (i.e. `gravatar.com/avatar/`)?

File: web/packages/quick-editor/src/quick-editor.ts
Comment (wellyshen) [reply web/packages/quick-editor/src/quick-editor.ts]: OK, I think [a required option](https://github.com/Automattic/gravatar/pull/14/files/38b6d671a01858bb0952c2a42bf340f1e62f3236#diff-ba766ac9a32a6d44162134bd2dab092d092e8789ba574e3150b4adf4fc22781dR5) should be fine.

Q: I noticed that the [avatarSelector](https://github.com/Automattic/gravatar/pull/14/files/38b6d671a01858bb0952c2a42bf340f1e62f3236#diff-ba766ac9a32a6d44162134bd2dab092d092e8789ba574e3150b4adf4fc22781dR6) option is also required. Do developers have to set it to use the library?

**Comment by wellyshen** [comment]
> I'm not sure we should allow a developer to close the popup. Imagine a user is using the Quick Editor, and out of nowhere, the popup closes because the developer called the close method. That would be a poor experience.

We could offer this feature and let developers decide how to use it in their specific cases. One more possible use case is that developers could use it to automatically close the popup once the user's avatar has been updated, instead of requiring the user to close it manually. However, if you think it's not necessary at this stage, I'm okay with leaving it out. We can add it when needed.

> I don't think that's the purpose of that class. If a developer wants more control, they could use the GravatarQuickEditorCore.

Let's validate it when we integrate the library with other products.

> Maybe that's something we could address in a follow-up PR?

No problems!

File: web/packages/quick-editor/src/quick-editor.ts
Comment (jcheringer) [reply web/packages/quick-editor/src/quick-editor.ts:35]: Checking the Gravatar URL would work in most of the scenarios.
But there's one specific scenario where it wouldn't work well:
Imagine a developer using their own default user profile image (not the Gravatar one) for users that don't have a defined avatar.
If they set this element to be updated after the user adds a profile image, the element wouldn't be updated because of the URL check.
It seems like an edge case, but it's not impossible to happen.

File: web/packages/quick-editor/webpack/config.core.js
Comment (jcheringer) [reply web/packages/quick-editor/webpack/config.core.js:60]: Yes, I did some tests, and the current Webback configuration causes it to be overridden.
However, I inspected the Webpack output configuration, and it seems the [assign-properties](https://webpack.js.org/configuration/output/#type-assign-properties) configuration does the merging.
In my tests, I was able to have both packages merged under the `Gravatar` scope.
I'm unfamiliar with the `UMD` purposes, so I might be missing some context.
Do you think using `assign-properties` would work?

File: web/packages/quick-editor/src/index.ts
Comment (jcheringer) [reply web/packages/quick-editor/src/index.ts]: I have changed it as you suggested above.

File: web/packages/quick-editor/src/quick-editor.ts
Comment (jcheringer) [reply web/packages/quick-editor/src/quick-editor.ts]: No, the `avatarSelector` isn't required. Good catch! ✅

File: web/packages/quick-editor/src/quick-editor-core.ts
Comment (jcheringer) [reply web/packages/quick-editor/src/quick-editor-core.ts:70]: I might have failed to understand your suggestion.
How would checking the `scope` before opening the popup help the developers guess the correct scopes?

File: web/packages/quick-editor/webpack/config.core.js
Comment (wellyshen) [reply web/packages/quick-editor/webpack/config.core.js:60]: I have not tried it but we can give it a try, if it doesn't work we can fix it in a new version.

File: web/packages/quick-editor/src/quick-editor-core.ts
Comment (wellyshen) [reply web/packages/quick-editor/src/quick-editor-core.ts:70]: Yep, my original thoughts are both:

- (Recommend) Check the incorrect value for developers and remind them by error logs
- (Good to have) Prevent the incorrect values are passed to gravatar.com

File: web/packages/quick-editor/src/quick-editor.ts
Comment (wellyshen) [reply web/packages/quick-editor/src/quick-editor.ts:35]: > Imagine a developer using their own default user profile image (not the Gravatar one) for users that don't have a defined avatar.
> If they set this element to be updated after the user adds a profile image, the element wouldn't be updated because of the URL check.

If that's the case, the avatar's URLs are different (e.g. `https://abc.com/default-avatar` v.s. `https://secure.gravatar.com/avatar/123`), so developers seem to have to update the URL by themself (by using the core module's event). Am I understanding correctly?

Comment (jcheringer) [reply web/packages/quick-editor/src/quick-editor.ts:35]: If we don't check the URL, developers wouldn't need to update the URL manually.
Honestly, I don't see a great benefit in checking the URL.

Comment (wellyshen) [reply web/packages/quick-editor/src/quick-editor.ts:35]: > If we don't check the URL, developers wouldn't need to update the URL manually.

I might be wrong. As I understand the above case:
 
- Developers use their default avatar image for an `img` element, e.g. `https://abc.com/default-avatar`
- The user updates their Gravatar's avatar. E.g. `https://secure.gravatar.com/avatar/123`
- Developers will need to change the URL from `https://abc.com/default-avatar` to `https://secure.gravatar.com/avatar/123`
- I don't think the `https://abc.com/default-avatar?t=xxxx` (through the `avatarSelector` option) could be helpful for this case

> Honestly, I don't see a great benefit in checking the URL.

I suppose the `avatarSelector` is used for Gravatar's URLs only. But if you think it's not worth checking the URLs. I'm fine without it.

File: web/packages/quick-editor/webpack/config.core.js
Comment (jcheringer) [reply web/packages/quick-editor/webpack/config.core.js:60]: Just to make sure you understood my idea:
For this option to work as expected, we would need to change both packages, `hovercards` and `quick-editor`, to use the `assign-properties`.

File: web/packages/quick-editor/src/quick-editor-core.ts
Comment (jcheringer) [reply web/packages/quick-editor/src/quick-editor-core.ts:70]: Addressed on [7b1ce7e](https://github.com/Automattic/gravatar/pull/14/commits/7b1ce7e61080364a77dbfd5e99a9f73208409360)

File: web/packages/quick-editor/src/quick-editor.ts
Comment (jcheringer) [reply web/packages/quick-editor/src/quick-editor.ts:35]: You're right, sorry!
In my mind, I was thinking we were receiving the avatar URL from the PostMessage, but we are not.
I'll add the check.
Sorry for the confusion.

Comment (jcheringer) [reply web/packages/quick-editor/src/quick-editor.ts:35]: Addressed on https://github.com/Automattic/gravatar/pull/14/commits/c0575eb75f05e9245b641d1fab5ffeac3ca43855

File: web/packages/quick-editor/webpack/config.core.js
Comment (wellyshen) [reply web/packages/quick-editor/webpack/config.core.js:60]: That's right bro! We can address this issue in the following PRs.

File: web/packages/quick-editor/src/quick-editor-core.ts
Comment (wellyshen) [inline web/packages/quick-editor/src/quick-editor-core.ts]: Since we have already assigned the type to the method, we can remove the duplicated arg type:

```suggestion
	open: Open = ( email ) => {
```

File: web/packages/quick-editor/src/quick-editor.ts
Comment (wellyshen) [inline web/packages/quick-editor/src/quick-editor.ts]: Nit: These variables are only used in the `constructor`. We can save some bytes for developers by using local scope variables. It's your call bro 😎.

**Review by wellyshen — APPROVED** [review]
@jcheringer Thank you for all the discussions and address all the feedback 💙. I think these modules are gooooood enough to move forward. I have left two more comments, please feel free to merge this PR when it's ready 🚀.

In addition, we can address [the UMD name scope issue](https://github.com/Automattic/gravatar/pull/14/files#r1662202020) in the following PRs.

File: web/packages/quick-editor/src/quick-editor-core.ts
Comment (jcheringer) [reply web/packages/quick-editor/src/quick-editor-core.ts]: ✅

File: web/packages/quick-editor/src/quick-editor.ts
Comment (jcheringer) [reply web/packages/quick-editor/src/quick-editor.ts]: ✅

**Pull request merged**
